src/hotspot/share/classfile/classFileParser.cpp

```diff
@@ -5299,6 +5299,9 @@ void ClassFileParser::verify_legal_field_signature(const Symbol* name,
                                                    const Symbol* signature,
                                                    TRAPS) const {
   if (!_need_verify) { return; }
+  if (!supports_inline_types() && signature->is_Q_signature()) {
```

Comment:
Shouldn't the test also include a check for Q-arrays?  (signature->is_Q_array_signature()) ?

Reply:
Yes, it should check for Q-arrays.  I'll add the check.  Thanks!